src/sentry/rules/actions/base.py

```diff
@@ -7,20 +7,29 @@
 
 from sentry.eventstore.models import GroupEvent
 from sentry.models.rule import Rule
+from sentry.models.rulefirehistory import RuleFireHistory
 from sentry.rules.base import CallbackFuture, EventState, RuleBase
 
 logger = logging.getLogger("sentry.rules")
 
 
-def instantiate_action(rule: Rule, action):
+def instantiate_action(rule: Rule, action, rule_fire_history: RuleFireHistory | None = None):
+    from sentry.integrations.slack import SlackNotifyServiceAction
     from sentry.rules import rules
 
-    action_cls = rules.get(action["id"])
+    action_id = action["id"]
+    action_cls = rules.get(action_id)
     if action_cls is None:
         logger.warning("Unregistered action %r", action["id"])
         return None
 
-    action_inst = action_cls(rule.project, data=action, rule=rule)
+    if action_id == SlackNotifyServiceAction.id:
+        action_inst = action_cls(
+            rule.project, data=action, rule=rule, rule_fire_history=rule_fire_history
+        )
+    else:
+        action_inst = action_cls(rule.project, data=action, rule=rule)
+
```

Comment:
Maybe I can leverage kwargs here, but is that any better?
```suggestion
    kwargs = {
        "data": action,
        "rule": rule,
    }
    if action_id == SlackNotifyServiceAction.id:
        kwargs["rule_fire_history"] = rule_fire_history
   
    action_inst = action_cls(rule.project, **kwargs)

```

Reply:
Hm! I don't see much difference to be honest since rule_filre_history should be passes sometime and not the other times it'll look a bit messy either way.